src/Features/CSharp/Portable/Braces/CSharpBracesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,404 @@
+                        forEachStatement.ForEachKeyword.GetLocation(), SyntaxFacts.GetText(SyntaxKind.ForEachKeyword));
+                }
+            }
+
+            if (node.IsKind(SyntaxKind.WhileStatement))
+            {
+                var whileStatement = (WhileStatementSyntax)node;
+                if (AnalyzeWhileStatement(whileStatement))
+                {
+                    return Diagnostic.Create(_severityToAddBracesDescriptorMap[severity],
+                        whileStatement.WhileKeyword.GetLocation(), SyntaxFacts.GetText(SyntaxKind.WhileKeyword));
+                }
+            }
+
+            if (node.IsKind(SyntaxKind.DoStatement))
+            {
+                var doStatement = (DoStatementSyntax)node;
+                if (AnalyzeDoStatement(doStatement))
+                {
+                    return Diagnostic.Create(_severityToAddBracesDescriptorMap[severity],
+                        doStatement.DoKeyword.GetLocation(), SyntaxFacts.GetText(SyntaxKind.DoKeyword));
+                }
+            }
+
+            if (node.IsKind(SyntaxKind.UsingStatement))
+            {
+                var usingStatement = (UsingStatementSyntax)node;
+                if (AnalyzeUsingStatement(usingStatement))
+                {
+                    return Diagnostic.Create(_severityToAddBracesDescriptorMap[severity],
+                        usingStatement.UsingKeyword.GetLocation(), SyntaxFacts.GetText(SyntaxKind.UsingKeyword));
+                }
+            }
+
+            if (node.IsKind(SyntaxKind.LockStatement))
+            {
+                var lockStatement = (LockStatementSyntax)node;
+                if (AnalyzeLockStatement(lockStatement))
+                {
+                    return Diagnostic.Create(_severityToAddBracesDescriptorMap[severity],
+                        lockStatement.LockKeyword.GetLocation(), SyntaxFacts.GetText(SyntaxKind.LockKeyword));
+                }
+            }
+
+            if (node.IsKind(SyntaxKind.FixedStatement))
+            {
+                var fixedStatement = (FixedStatementSyntax)node;
+                if (AnalyzeFixedStatement(fixedStatement))
+                {
+                    return Diagnostic.Create(_severityToAddBracesDescriptorMap[severity],
+                        fixedStatement.FixedKeyword.GetLocation(), SyntaxFacts.GetText(SyntaxKind.LockKeyword));
+                }
+            }
+
+            return null;
+        }
+
+        private Diagnostic GetRemoveBracesDiagnostic(SyntaxNode node, DiagnosticSeverity severity)
+        {
+            if (node.IsKind(SyntaxKind.IfStatement))
+            {
+                var ifStatement = (IfStatementSyntax)node;
+                if (AnalyzeBlockedIfStatement(ifStatement))
+                {
+                    return Diagnostic.Create(_severityToRemoveBracesDescriptorMap[severity],
+                        ifStatement.IfKeyword.GetLocation(), SyntaxFacts.GetText(SyntaxKind.IfKeyword));
```

Comment:
lots of redundancy here and below.  We can make a helper to do this given the severity and the token to report on.